nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/CaptureBoxEvents.java

```diff
@@ -0,0 +1,237 @@
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicLong;
+
+@PrimaryNodeOnly
+@TriggerSerially
+@Tags({"box", "storage"})
+@CapabilityDescription("""
+        Captures all events from Box. This processor can be used to capture events such as uploads, modifications, deletions, etc.
+        The content of the events is sent to the 'success' relationship as a JSON array.
+        """)
+@SeeAlso({ FetchBoxFile.class, PutBoxFile.class, ListBoxFile.class })
+@InputRequirement(Requirement.INPUT_FORBIDDEN)
+public class CaptureBoxEvents extends AbstractProcessor implements VerifiableProcessor {
+
+    public static final PropertyDescriptor MAX_MESSAGE_QUEUE_SIZE = new PropertyDescriptor.Builder()
+            .name("Queue Capacity")
+            .description("""
+                    The maximum size of the internal queue used to buffer events being transferred from the underlying stream to the processor.
+                    Setting this value higher allows more messages to be buffered in memory during surges of incoming messages, but increases the total
+                    memory used by the processor during these surges.
+                    """)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("10000")
+            .required(true)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = List.of(
+            BoxClientService.BOX_CLIENT_SERVICE,
+            MAX_MESSAGE_QUEUE_SIZE
+    );
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Events received successfully will be sent out this relationship.")
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_SUCCESS);
+
+    private volatile BoxAPIConnection boxAPIConnection;
+    private volatile EventStream eventStream;
+    protected volatile LinkedBlockingQueue<BoxEvent> events;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public final Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        final BoxClientService boxClientService = context.getProperty(BoxClientService.BOX_CLIENT_SERVICE).asControllerService(BoxClientService.class);
+        boxAPIConnection = boxClientService.getBoxApiConnection();
+        eventStream = new EventStream(boxAPIConnection);
```

Comment:
This constructor starts processing events at the current time according to the [EventStream](https://github.com/box/box-java-sdk/blob/f6c37ce61ed02eb21ce6eb6ec5d23ffcabbb76d2/src/main/java/com/box/sdk/EventStream.java#L46) class constructor.

This seems like a reasonable initial implementation, but are there any concerns around picking up from the last position on start or stop using state information?

Reply:
I didn't even notice that it was possible to resume at a specific position, things make much more sense now. I've changed the code to not clear the queue if the processor is stopped and reuse the queue if restarted, this way stopping/starting the processor would not cause missing events. It could still happen in case of a restart though.